qualtran/bloqs/arithmetic/comparison.py

```diff
@@ -1154,3 +1156,423 @@ def _clineardepthgreaterthan_example() -> CLinearDepthGreaterThan:
+            c = add_ints(-int(a), int(b), num_bits=self.dtype.bitsize + 1, is_signed=False)
+        else:
+            c = add_ints(int(a), -int(b), num_bits=self.dtype.bitsize + 1, is_signed=False)
+        return {'a': a, 'b': b, 'c': c, 'target': int(self._classical_comparison(a, b))}
+
+    def _compute(self, bb: 'BloqBuilder', a: 'Soquet', b: 'Soquet') -> Dict[str, 'SoquetT']:
+        if self.comparison_dtype in ('>', '<='):
+            a, b, c, target = bb.add_from(self._half_greater_than_bloq, a=a, b=b)  # type: ignore
+        else:
+            b, a, c, target = bb.add_from(self._half_greater_than_bloq, a=b, b=a)  # type: ignore
+
+        if self.comparison_dtype in ('<=', '>='):
+            target = bb.add(XGate(), q=target)
+
+        return {'a': a, 'b': b, 'c': c, 'target': target}
+
+    def _uncompute(
+        self, bb: 'BloqBuilder', a: 'Soquet', b: 'Soquet', c: 'Soquet', target: 'Soquet'
+    ) -> Dict[str, 'SoquetT']:
+        if self.comparison_dtype in ('<=', '>='):
+            target = bb.add(XGate(), q=target)
+
+        if self.comparison_dtype in ('>', '<='):
+            a, b = bb.add_from(self._half_greater_than_bloq.adjoint(), a=a, b=b, c=c, target=target)  # type: ignore
+        else:
+            a, b = bb.add_from(self._half_greater_than_bloq.adjoint(), a=b, b=a, c=c, target=target)  # type: ignore
+
+        return {'a': a, 'b': b}
+
+    def build_composite_bloq(
+        self,
+        bb: 'BloqBuilder',
+        a: 'Soquet',
+        b: 'Soquet',
+        c: Optional['Soquet'] = None,
+        target: Optional['Soquet'] = None,
+    ) -> Dict[str, 'SoquetT']:
+        if self.uncompute:
+            assert c is not None
+            assert target is not None
+            return self._uncompute(bb, a, b, c, target)
+        else:
+            assert c is None
+            assert target is None
+            return self._compute(bb, a, b)
+
+    def build_call_graph(self, ssa: 'SympySymbolAllocator') -> 'BloqCountDictT':
+        extra_ops = {}
+        if isinstance(self.dtype, QInt):
+            extra_ops = {
+                SignExtend(self.dtype, QInt(self.dtype.bitsize + 1)): 2,
+                SignExtend(self.dtype, QInt(self.dtype.bitsize + 1)).adjoint(): 2,
+            }
+        if self.comparison_dtype in ('>=', '<='):
+            extra_ops[XGate()] = 1
+        adder = self._half_greater_than_bloq
+        if self.uncompute:
+            adder = adder.adjoint()
+        adder_call_graph = adder.build_call_graph(ssa)
+        assert isinstance(adder_call_graph, dict)
+        counts = {**adder_call_graph}
+        for k, v in extra_ops.items():
+            if k not in counts:
+                counts[k] = 0
+            counts[k] += v
+        return counts
+
+
+@frozen
+class _LinearDepthHalfComparisonBase(_HalfComparisonBase):
+    @cached_property
+    def _half_greater_than_bloq(self) -> Bloq:
+        return _HalfLinearDepthGreaterThan(self.dtype, uncompute=False)
```

Comment:
is this always going to be the value used here? why even factor it out at all?

Reply:
same as above ... it's a property of the LinearDepth variant but we might want to add the _LogDepth variant